tests/tensorboard_test.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for flax.metrics.tensorboard."""
+import itertools
+import tempfile
+
+from absl.testing import absltest
+import numpy as onp
+
+from tensorboard.backend.event_processing import directory_watcher
+from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.util import tensor_util
+import tensorflow.compat.v2 as tf
+
+from flax.metrics.tensorboard import SummaryWriter
+
+def _process_event(event):
+  for value in event.summary.value:
+    yield {'wall_time': event.wall_time, 'step': event.step, 'value': value}
+
+
+class TensorboardTest(absltest.TestCase):
+
+  def parse_and_return_summary_value(self, path):
+    """Parse the event file in the given path and return the
+    only summary value."""
+    event_value_list = []
+    event_file_generator = directory_watcher.DirectoryWatcher(
+        path, event_file_loader.EventFileLoader).Load()
+    event_values = itertools.chain.from_iterable(
+        map(_process_event, event_file_generator))
+    for value_dict in event_values:
+      event_value_list.append(value_dict)
+
+    self.assertLen(event_value_list, 1)
+    self.assertEqual(event_value_list[0]['step'], 1)
+    self.assertGreater(event_value_list[0]['wall_time'], 0.0)
+    return event_value_list[0]['value']
+
+  def assert_encoded_audio_files_equal(self, actual_audio, expected_audio):
```

Comment:
I am somewhat in favour of inlining assert and trim functions form this method into the actual tests using this helper method. It would not duplicate a lot of code, but make the tests easier to read.

Reply:
Yeah I think that makes sense. Inlined the necessary assertions. Thanks!